UndertaleModTool/Scripts/Resource Unpackers/ExportSpecificSprites.csx

```diff
@@ -0,0 +1,70 @@
+//Basically a mashup of ExportAllSprites and DumpSpecificCode
+
+using System.Text;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using UndertaleModLib.Util;
+
+EnsureDataLoaded();
+
+bool padded = ScriptQuestion("Export all sprites WITH padding?");
+
+string texFolder = PromptChooseDirectory();
+if (texFolder == null)
+    throw new ScriptException("The export folder was not set.");
+texFolder = texFolder + "Sprites" + Path.DirectorySeparatorChar;
+Directory.CreateDirectory(texFolder);
+
+TextureWorker worker = new TextureWorker();
+
+List<UndertaleSprite> spritesToDump = new List<UndertaleSprite>();
+List<String> splitStringsList = new List<String>();
+
+string inputtedText = "";
+inputtedText = SimpleTextInput("Menu", "Enter the name of the sprites", "", true);
+string[] individualLineArray = inputtedText.Split('\n', StringSplitOptions.RemoveEmptyEntries);
+foreach (var oneLine in individualLineArray)
+{
+    splitStringsList.Add(oneLine.Trim());
+}
+foreach (string listElement in splitStringsList)
+{
+    foreach (UndertaleSprite spr in Data.Sprites)
+    {
+        if (listElement.ToLower() == spr.Name.Content.ToLower())
+        {
+            spritesToDump.Add(spr);
+        }
+    }
+}
+
+SetProgressBar(null, "Sprites", 0, spritesToDump.Count);
+StartProgressBarUpdater();
+
+
+
+await Task.Run(() => {
+    foreach (UndertaleSprite sprToDump in spritesToDump)
+    {
+        DumpSprite(sprToDump);
+    }
+});
+
+worker.Cleanup();
+await StopProgressBarUpdater();
+HideProgressBar();
+ScriptMessage("Export Complete.\n\nLocation: " + texFolder);
+
+void DumpSprite(UndertaleSprite sprite)
+{
+    for (int i = 0; i < sprite.Textures.Count; i++)

```

Comment:
you missed converting this to a foreach

Reply:
How can I turn it into a foreach? I was thinking to use
`foreach(... Textures in sprite)`
but i have no idea of what data type to use, in other words I don't know how to write this foreach

Reply:
Disregard this, I just noticed you actually do need the index later down the line.

----
But if you were curious on how you would've written the foreach: the data type is always the one of the Enumerable you're enumerating through. Here, you were looping through `sprite.Textures`.  If you [look at the definition of sprite.Textures](https://github.com/krzys-h/UndertaleModTool/blob/master/UndertaleModLib/Models/UndertaleSprite.cs#L196) you can see that it is a `UndertaleSimpleList<UndertaleSprite.TextureEntry>` So you `foreach (UndertaleSprite.TextureEntry texture in sprite.Textures)`.

Alternatively, you also could've used the `var` keyword for data types that are too long/bothersome to type out: `foreach (var texture in sprite.Textures)`